.github/workflows/test-CI.yml

```diff
@@ -0,0 +1,166 @@
+
+      # A bunch of if-else. Might move to an action
+      # Decide environment based on the target branch (for both push and PR events)
+      - name: Set environment based on target branch
+        run: |
+          set -ex
+          if [[ "${{ github.event_name }}" == "pull_request_target" ]]; then
+            # Use the target branch of the pull request
+            echo "were in pull_request_target"
+            target_branch="${{ github.event.pull_request.base.ref }}" # Probably should be updated to produce the target branch and not the branch
+          elif [[ "${{ github.event_name }}" == "push" ]]; then
+            echo "were in push"
+            target_branch="$(gh pr view --json baseRefName,headRefName --jq .baseRefName)"
+          else
+            # Use the branch of the push event
+            # todo: verify that dispatch works
+            echo "On dsipatch - WIP"
+            target_branch="${{ github.ref_name }}"
+            echo $target_branch
+            echo $target_branch
+            echo $target_branch
+            echo $target_branch
+            echo $target_branch
+          fi
+
+          if [[ "$target_branch" == "main" ]]; then
+            echo "Running on prod environment."
+
+            echo "M2M_SECRET_ARN=${{ secrets.PROD_M2M_SECRET_ARN }}" >> $GITHUB_ENV
+
+            echo "CLASSIQ_IDE=https://platform.classiq.io" >> $GITHUB_ENV
+            echo "CLASSIQ_HOST=https://api.classiq.io" >> $GITHUB_ENV
+            echo "IS_DEV=false" >> $GITHUB_ENV
+          else
+            echo "Running on dev environment."
+
+            echo "M2M_SECRET_ARN=${{ secrets.NIGHTLY_M2M_SECRET_ARN }}" >> $GITHUB_ENV
+
+            echo "CLASSIQ_IDE=https://nightly.platform.classiq.io" >> $GITHUB_ENV
+            echo "CLASSIQ_HOST=https://staging.api.classiq.io" >> $GITHUB_ENV
+            echo "IS_DEV=true" >> $GITHUB_ENV
+          fi
+        shell: bash
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      # The following 2 steps can also be grouped into one action
+      # Step to detect changed .ipynb files
+      - name: Get changed notebook files
+        id: changed-files-ipynb
+        uses: tj-actions/changed-files@v44
+        with:
+          files: |
+            **/*.ipynb
+
+      - name: Print changed notebook files
+        run: |
+          echo "Changed notebook files: ${{ steps.changed-files-ipynb.outputs.all_changed_files }}"
+
+      - name: Set changed notebook into environment variables
+        run: |
+          set -ex
+          if [ "${{ github.event_name }}" == 'pull_request_target' ]; then
+            echo "SHOULD_TEST_ALL_FILES=false" >> $GITHUB_ENV
+          elif [[ "${{ github.event_name }}" == 'workflow_dispatch' || "${{ github.head_ref || github.ref_name }}" == "dev" ]]; then
+            echo "SHOULD_TEST_ALL_FILES=true" >> $GITHUB_ENV
+          fi
+
+      - name: Install dependencies
+        run: |
+          set -e
+          # Pre is needed for Dev pre releases
```

Comment:
document

Reply:
I am unsure myself,I probably took it as it is from a script by Zach/ Dor.  As far I understand, it's needed for using pre-release versions. I have clarified it.